test/components/admin/datetime_component_test.rb

```diff
@@ -0,0 +1,137 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+class Admin::DatetimeComponentTest < ViewComponent::TestCase
+  test "builds the name of the select inputs based on the class of the object" do
+    consultation = create(:consultation)
+
+    render_inline(
+      Admin::DatetimeComponent.new(
+        object: consultation,
+        field_name: "first_published_at",
+      ),
+    )
+
+    assert_selector "select[name='consultation[first_published_at(1i)]']"
+    assert_selector "select[name='consultation[first_published_at(2i)]']"
+    assert_selector "select[name='consultation[first_published_at(3i)]']"
+    assert_selector "select[name='consultation[first_published_at(4i)]']"
+    assert_selector "select[name='consultation[first_published_at(5i)]']"
+  end
+
+  test "overrides the name of the select inputs with a prefix if one is passed in" do
+    consultation = create(:consultation)
+
+    render_inline(
+      Admin::DatetimeComponent.new(
+        object: consultation,
+        field_name: "first_published_at",
+        prefix: "edition",
+      ),
+    )
+
+    assert_selector "select[name='edition[first_published_at(1i)]']"
+    assert_selector "select[name='edition[first_published_at(2i)]']"
+    assert_selector "select[name='edition[first_published_at(3i)]']"
+    assert_selector "select[name='edition[first_published_at(4i)]']"
+    assert_selector "select[name='edition[first_published_at(5i)]']"
+  end
+
+  test "renders a blank set of datetime inputs when the relevant datetime field on the object passed in is nil" do
+    edition = build_stubbed(:edition, first_published_at: nil)
+    render_inline(
+      Admin::DatetimeComponent.new(
+        object: edition,
+        field_name: "first_published_at",
+      ),
+    )
+
+    assert_equal page.all("h3")[0].text, "Date"
+    assert_equal page.all("label")[0].text, "Day"
```

Comment:
Would suggest you don't need these since this isn't code affected by the inputs to the component or is it the scenario described above.

These don't seem things that are worth testing anyway since they essentially are just confirming that you typed the same header and label in your HTML and aren't controlled by logic.